RELEASE_NOTES.md

```diff
@@ -8,6 +8,27 @@
 
 - The `BatteryPool.power_status` method now streams objects of type `BatteryPoolReport`.  They're identical to the previous `Report` objects, except for the name of the class.
 
+- `microgrid.grid()`
+  - A `Grid` object will always be instantiated now, even if the microgrid is not connected to the grid (islanded microgrid).
+  - The fuse will be set to `None` to indicate that the grid connection point is not available, either because the fuse is unknown (not configured) or because the microgrid is islanded.
```

Comment:
I'm not sure if I'm forgetting something or if there is a technical difficulty to implement it differently, but wouldn't it make more sense to still set a fuse with a rate of 0 for islands, as there could never be any electricity flow through the grid connection in this case?

Reply:
I understood `Grid` should be agnostic to the "type" of the microgrid and shouldn't make any difference whether the microgrid is islanded or not. The user can always check the component graph if needs to know there is a `grid` component or not. I tried to state the most known examples in the release notes but it seems this is misleading in the sense that makes you think the Grid.fuse can be used to somehow classify the microgrid

Reply:
> I understood Grid should be agnostic to the "type" of the microgrid and shouldn't make any difference whether the microgrid is islanded or not.

Yes, me too, I think we are on the same page with this one.

> The user can always check the component graph if needs to know there is a grid component or not.

Also agreed.

> I tried to state the most known examples in the release notes but it seems this is misleading in the sense that makes you think the Grid.fuse can be used to somehow classify the microgrid

I think there are 2 issues here:

1. The release note is misleading, specifically "The fuse will be set to `None` to indicate that the grid connection point is not available". This is not true, right? If `Fuse` is `None` is because the `Fuse` component / metadata (not sure if it came from the metadata or not) is not present in the microgrid API response, correct?

2. For applications to be able to use the `.grid()` in the most general way, I think it would make sense to create a fuse with a rate of 0 for islands instead of leaving the `Fuse` as `None`. This is because if we leave it at `None`, then users needs to do a special case for unknown fuse state, so they will probably need to check the component graph to see if there is a `Grid` component or not, and then if there is not, say "OK, then no current can flow through the grid" which is effectively the same as having a fuse of 0, no electricity can flow (or the fuse will be blown). So any code dealing with **existing** fuses, will behave correctly for islands without having to go check out if we are on an island or not.

I think it might be even good to have a `is_connected()` method in `Grid` as a shortcut to find out if we are connected to the grid or not (but that's for another PR/time).

Reply:
> If Fuse is None is because the Fuse component / metadata (not sure if it came from the metadata or not) is not present in the microgrid API response, correct?

That's right

> For applications to be able to use the .grid() in the most general way, I think it would make sense to create a fuse with a rate of 0 for islands instead of leaving the Fuse as None. This is because if we leave it at None, then users needs to do a special case for unknown fuse state, so they will probably need to check the component graph to see if there is a Grid component or not, and then if there is not, say "OK, then no current can flow through the grid" which is effectively the same as having a fuse of 0, no electricity can flow (or the fuse will be blown). So any code dealing with existing fuses, will behave correctly for islands without having to go check out if we are on an island or not.

These are good points and make sense to me. Somehow I got confused when we previously discussed about it. I'll amend the PR accordingly.

> I think it might be even good to have a is_connected() method in Grid as a shortcut to find out if we are connected to the grid or not (but that's for another PR/time).

I'll create an issue for this

Reply:
Updated to set the rated current of the grid fuse to `Current.zero` in case of a microgrid configured as an island.
Also updated the release notes.

Reply:
Also created https://github.com/frequenz-floss/frequenz-sdk-python/issues/787

Reply:
> These are good points and make sense to me. Somehow I got confused when we previously discussed about it. I'll amend the PR accordingly.

OK, @shsms @matthias-wende-frequenz  FYI, just in case I misunderstood anything myself. It would be good to confirm we all think this is the best approach.

Reply:
This looks like the right approach, saying `None` only when there insufficient info,  and we're not even having to do a separate lookup of the component graph, so that's great as well!